### ISSUE TITLE
Usage of torch.sum in TransD implementation

### ISSUE THREAD

**Issue by apoorvb** [thunlp/OpenKE#234]
Can you explain why is torch.sum used in the following function?
`
def _transfer(self, e, e_transfer, r_transfer):
		if e.shape[0] != r_transfer.shape[0]:
			e = e.view(-1, r_transfer.shape[0], e.shape[-1])
			e_transfer = e_transfer.view(-1, r_transfer.shape[0], e_transfer.shape[-1])
			r_transfer = r_transfer.view(-1, r_transfer.shape[0], r_transfer.shape[-1])
			e = F.normalize(
				self._resize(e, -1, r_transfer.size()[-1]) + torch.sum(e * e_transfer, -1, True) * r_transfer,
				p = 2, 
				dim = -1
			)			
			return e.view(-1, e.shape[-1])
		else:
			return F.normalize(
				self._resize(e, -1, r_transfer.size()[-1]) + torch.sum(e * e_transfer, -1, True) * r_transfer,
				p = 2, 
				dim = -1
			)
`

Why is the last dimension summed with torch.sum instead of multiplying directly using '*' operator?

Additionally, in transD for calculation of entity transfer if I compute(get product of ent.transfer and relation transfer then add with identity matrix) internal terms of projection matrix and then multiply with entity embedding, what would be mathematically/theoretically wrong with it? and would torch.sum still be required anywhere?

**Comment by THUCSTHanxu13** [comment]
'A * B' is not the inner product for embeddings A and B, and it is just multiplies embeddings A and B by multiplying their corresponding elements. Hence, we adopt 'torch.sum' to compute the inner product for embeddings.
